cmd/node/app_test.go

```diff
@@ -92,73 +70,76 @@ func Test_PoETHarnessSanity(t *testing.T) {
+	name := 'a'
+	for i := 0; i < numOfInstances; i++ {
+		dbStorepath := storeFormat + string(name)
+		smApp, err := InitSingleInstance(*cfg, i, genesisTime, rng, dbStorepath, rolacle, poetClient, fastHare, clock)
+		assert.NoError(suite.T(), err)
+		suite.apps = append(suite.apps, smApp)
+		suite.dbs = append(suite.dbs, dbStorepath)
+		name++
+	}
+}
 
-	smApp.Config.HARE.N = 5
-	smApp.Config.HARE.F = 2
-	smApp.Config.HARE.RoundDuration = 3
-	smApp.Config.HARE.WakeupDelta = 5
-	smApp.Config.HARE.ExpectedLeaders = 5
-	smApp.Config.CoinbaseAccount = strconv.Itoa(i + 1)
-	smApp.Config.LayerAvgSize = 5
-	smApp.Config.LayersPerEpoch = 3
-	smApp.Config.Hdist = 5
-	smApp.Config.GenesisTime = genesisTime
-	smApp.Config.LayerDurationSec = 20
-	smApp.Config.HareEligibility.ConfidenceParam = 4
-	smApp.Config.HareEligibility.EpochOffset = 0
-	smApp.Config.StartMining = true
-	smApp.Config.SyncRequestTimeout = 2000
+var tests = []TestScenario{txWithRunningNonceGenerator([]int{}), sameRootTester([]int{0}), reachedEpochTester([]int{}), txWithUnorderedNonceGenerator([]int{1})}
 
-	smApp.Config.API = apiCfg.DefaultConfig()
+func (suite *AppTestSuite) TestMultipleNodes() {
+	//EntryPointCreated <- true
+	const numberOfEpochs = 5 // first 2 epochs are genesis
+	//addr := address.BytesToAddress([]byte{0x01})
+	cfg := getTestDefaultConfig()
 
-	edSgn := signing.NewEdSigner()
-	pub := edSgn.PublicKey()
+	path := "../tmp/test/state_" + time.Now().String()
 
-	vrfPriv, vrfPub := BLS381.GenKeyPair(rng)
-	vrfSigner := BLS381.NewBlsSigner(vrfPriv)
-	nodeID := types.NodeId{Key: pub.String(), VRFPublicKey: vrfPub}
+	genesisTime := time.Now().Add(20 * time.Second).Format(time.RFC3339)
+	poetClient, err := NewRPCPoetHarnessClient()
+	if err != nil {
+		log.Panic("failed creating poet client harness: %v", err)
+	}
+	suite.poetCleanup = poetClient.Teardown
 
-	swarm := net.NewNode()
-	dbStorepath := storeFormat + name
+	rolacle := eligibility.New()
+	rng := BLS381.DefaultSeed()
 
-	hareOracle := oracle.NewLocalOracle(rolacle, 5, nodeID)
-	hareOracle.Register(true, pub.String())
+	gTime, err := time.Parse(time.RFC3339, genesisTime)
+	if err != nil {
+		log.Error("cannot parse genesis time %v", err)
+	}
+	ld := time.Duration(20) * time.Second
+	clock := timesync.NewTicker(timesync.RealClock{}, ld, gTime)
+	suite.initMultipleInstances(cfg, rolacle, rng, 5, path, genesisTime, poetClient, false, clock)
+	for _, a := range suite.apps {
+		a.startServices()
+	}
 
-	postClient, err := activation.NewPostClient(&smApp.Config.POST, util.Hex2Bytes(nodeID.Key))
-	r.NoError(err)
-	r.NotNil(postClient)
+	ActivateGrpcServer(suite.apps[0])
 
-	err = smApp.initServices(nodeID, swarm, dbStorepath, edSgn, false, hareOracle, uint32(smApp.Config.LayerAvgSize), postClient, poetClient, vrfSigner, uint16(smApp.Config.LayersPerEpoch))
-	r.NoError(err)
-	smApp.setupGenesis()
+	if err := poetClient.Start("127.0.0.1:9091"); err != nil {
```

Comment:
just a note, poet api might have changed. you should revisit it before merging

Reply:
The `go-spacemesh` repo depends on a specific version of the poet. Unit tests don't use the latest, this is a problem in automation tests only. The only reason this would break is if someone upgrades `go-spacemesh` before this PR is merged, and it will cause bors to block merging, since tests will fail.